unittest/server/src/ivi_input_controller_uinttests.cpp

```diff
@@ -0,0 +1,1714 @@
+    custom_wl_list_insert(&mpp_ctxSeat[0]->keyboard_grab.keyboard->focus_resource_list, &l_wlResource.link);
+    custom_wl_list_init(&mpp_ctxSeat[0]->keyboard_grab.keyboard->resource_list);
+
+    keyboard_grab_cancel(&mpp_ctxSeat[0]->keyboard_grab);
+
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 2);
+
+    free(l_surf[0]->resource);
+    free(l_surf[0]);
+    free(mpp_ctxSeat[0]->keyboard_grab.keyboard);
+}
+
+/** ================================================================================================
+ * @test_id             pointer_grab_focus_wrongButtonCount
+ * @brief               Test case of pointer_grab_focus() where input ctxSeat button_count is enable {1}
+ * @test_procedure Steps:
+ *                      -# Set input ctxSeat
+ *                      -# Calling the pointer_grab_focus()
+ *                      -# Verification point:
+ *                         +# surface_get_weston_surface() not be called
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(InputControllerTest, pointer_grab_focus_wrongButtonCount)
+{
+    mpp_ctxSeat[0]->pointer_grab.pointer = (struct weston_pointer *)malloc(sizeof(struct weston_pointer));
+    mpp_ctxSeat[0]->pointer_grab.pointer->button_count = 1;
+
+    pointer_grab_focus(&mpp_ctxSeat[0]->pointer_grab);
+
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 0);
+
+    free(mpp_ctxSeat[0]->pointer_grab.pointer);
+}
+
+/** ================================================================================================
+ * @test_id             pointer_grab_focus_nullForcedPtr
+ * @brief               Test case of pointer_grab_focus() where input ctxSeat button_count is disable {0}
+ *                      and forced_ptr_focus_surf is null pointer
+ * @test_procedure Steps:
+ *                      -# Set input ctxSeat (Do not set forced_ptr_focus_surf)
+ *                      -# Calling the pointer_grab_focus()
+ *                      -# Verification point:
+ *                         +# surface_get_weston_surface() not be called
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(InputControllerTest, pointer_grab_focus_nullForcedPtr)
+{
+    mpp_ctxSeat[0]->pointer_grab.pointer = (struct weston_pointer *)malloc(sizeof(struct weston_pointer));
+    mpp_ctxSeat[0]->pointer_grab.pointer->button_count = 0;
+    mpp_ctxSeat[0]->pointer_grab.pointer->seat = (struct weston_seat *)malloc(sizeof(struct weston_seat));
+    mpp_ctxSeat[0]->pointer_grab.pointer->seat->compositor = (struct weston_compositor *)0xFFFFFFFF;
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus = (struct weston_view *)malloc(sizeof(struct weston_view));
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus->surface = (struct weston_surface *)0xFFFFFFFF;
+
+    pointer_grab_focus(&mpp_ctxSeat[0]->pointer_grab);
+
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 0);
```

Comment:
forced_ptr_focus_surf is null, the logic will call to input_ctrl_ptr_set_west_focus(). need to add more verification points